source/verus/src/main.rs

```diff
@@ -75,19 +63,270 @@ fn main() {
+        std::mem::forget(job);
+    }
+
+    let reports = match reports {
+        Some(reports) => {
+            // proj path is only created after the project has --local-store flag
+            if !reports.proj_path.is_dir() {
+                cmd.status().map_err(|x| format!("verus failed to run with error: {}", x))?;
+                return Ok(());
+            }
+            reports
+        }
+        None => {
+            cmd.status().map_err(|x| format!("verus failed to run with error: {}", x))?;
+            return Ok(());
+        }
+    };
+
+    let proj_path = reports.proj_path;
+
+    cmd.arg(format!("--emit=dep-info={}", reports.dep_path.display()));
+
+    // clean all files in proj_path
+    remove_dir_all(&proj_path).expect("failed to remove repo_path");
+
+    let toml_path = proj_path.join("reports.toml");
+
+    create_dir_all(&proj_path).expect("creating reports.toml");
+    let mut file = File::create(toml_path).expect("creating reports.toml");
+
+    let mut child = cmd
+        .stdout(std::process::Stdio::piped())
+        .stderr(std::process::Stdio::piped())
+        .spawn()
+        .map_err(|x| format!("verus failed to run with error: {}", x))?;
+
+    let out = BufReader::new(child.stdout.take().expect("create BufReader for stdout"));
+    let err = BufReader::new(child.stderr.take().expect("create BufReader for stderr"));
+
+    let thread_err = std::thread::spawn(move || {
+        let mut stderr_output = String::new();
+        err.lines().for_each(|line| {
+            let line = line.unwrap();
+            eprintln!("{}", line);
+            let x = anstream::adapter::strip_str(&line).to_string() + "\n";
+            stderr_output.push_str(&x);
+        });
+        return stderr_output;
+    });
+
+    let thread_out = std::thread::spawn(move || {
+        let mut stdout_output = String::new();
+        out.lines().for_each(|line| {
+            let line = line.unwrap();
+            println!("{}", line);
+            let x = anstream::adapter::strip_str(&line).to_string() + "\n";
+            stdout_output.push_str(&x);
+        });
+        return stdout_output;
+    });
+
+    let err_string = thread_err.join().unwrap();
+    let out_string = thread_out.join().unwrap();
+
+    child.wait().map_err(|x| format!("verus failed to run with error: {}", x))?;
+
+    writeln!(file, "{}", err_string)
+        .map_err(|x| format!("failed to write to reports.toml: {}", x))?;
+    writeln!(file, "{}", out_string)
+        .map_err(|x| format!("failed to write to reports.toml: {}", x))?;
+
+    let dep_file_path = reports.dep_path;
+
+    let deps = get_dependencies(&dep_file_path)?;
+
+    let deps_path = deps.iter().map(|x| Path::new(x)).collect::<Vec<_>>();
+
+    // compute common ancester
+    let common = match common_path::common_path_all(deps_path) {
+        Some(common) => common,
+        None => PathBuf::from(""),
+    };
```

Comment:
It is _extremely_ unlikely that you can't find a common path to dependencies (the only situation I can think of where this can happen is if you are on Windows, and are using many separate drives for your dependencies), and if this ever happens, it is better to be loud about it, rather than silently replace it with the empty string.

In general, it is good to be loud about horrifying and rare cases, rather than silently replace them with _potentially_ valid information, making detection and debugging harder.

```suggestion
    let common = common_path::common_path_all(deps_path)
        .expect("Getting common ancestor for dependencies");
```